### PR TITLE
fix internal audit 1.5.0 + add role: position manager

### PR DESCRIPTION
**Pull request by DylanYang0523**
### PR Reminders
---

1. **Contract**: make sure the code follows our convention, ex: naming, explicit returns, etc.; if uncertain, discuss with others

2. **Test**: make sure tests can cover most normal and edge cases; if not, open follow-up tickets. Also, look out for failed tests and fix them!

3. **CHANGELOG.md**: update when external interfaces are changed

4. Workflow: 
    - Github: assign the pr to yourself; if pairing, can merge directly; else, assign someone to help review
    - Asana: assign the corresponding ticket to yourself and leave the pr link on it for easier follow-ups
    - Discord: if someone is mentioned in this pr, tag on Discord

### CODE REVIEW

File: test/foundry/OtcMaker/OtcMaker.setter.t.sol [hunk 1]
```diff
@@ -8,10 +8,19 @@ import { OtcMakerSetup } from "./helper/OtcMakerSetup.sol";
...
     using PerpSafeCast for uint24;
 
+    event CallerUpdated(address oldCaller, address newCaller);
+    event FundOwnerUpdated(address oldFundOwner, address newFundOwner);
+    event PositionManagerUpdated(address oldPositionManager, address newPositionManager);
+
     function test_set_caller() public {
         assertFalse(otcMaker.getCaller() == alice);
+
+        vm.expectEmit(true, false, false, true);
```
Comment (CheshireCatNick) [inline test/foundry/OtcMaker/OtcMaker.setter.t.sol:18]: check how to set this.

File: contracts/otcMaker/OtcMaker.sol [hunk 2]
```diff
@@ -38,33 +41,61 @@ contract OtcMaker is SafeOwnable, EIP712Upgradeable, IOtcMaker, OtcMakerStorageV
...
+    function setPositionManager(address newPositionManager) external onlyOwner {
+        _requireNonZeroAddress(newPositionManager);
+        address oldPositionManager = _positionManager;
+        _positionManager = newPositionManager;
+        emit PositionManagerUpdated(oldPositionManager, newPositionManager);
     }
 
-    function setMarginRatioLimit(uint24 marginRatioLimitArg) external override onlyOwner {
+    function setMarginRatioLimit(uint24 marginRatioLimitArg) external onlyOwner {
         require(marginRatioLimitArg > 62500 && marginRatioLimitArg < 1000000, "OM_IMR");
```
Comment (tailingchen) [inline contracts/otcMaker/OtcMaker.sol:93]: To prevent hardcode `62500`, suggest to query this value from ClearingHouseConfig.
```
uint24 mmRatio = IClearingHouseConfig(IClearingHouse(_clearingHouse).getClearingHouseConfig()).getMmRatio();

require(marginRatioLimitArg > mmRatio && marginRatioLimitArg < 1000000, "OM_IMR");
```

File: contracts/otcMaker/OtcMaker.sol [hunk 3]
```diff
@@ -4,6 +4,7 @@ pragma abicoder v2;
 
 import { SafeERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/SafeERC20Upgradeable.sol";
 import { IERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/SafeERC20Upgradeable.sol";
```
Comment (tailingchen) [inline contracts/otcMaker/OtcMaker.sol:6]: import { IERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";

File: contracts/otcMaker/OtcMaker.sol [hunk 4]
```diff
@@ -137,7 +185,7 @@ contract OtcMaker is SafeOwnable, EIP712Upgradeable, IOtcMaker, OtcMakerStorageV
         uint256 week,
         uint256 claimedBalance,
         bytes32[] calldata _merkleProof
```
Comment (tailingchen) [inline contracts/otcMaker/OtcMaker.sol:187]: remove `_` to keep naming consistent

**Review by paco0x — APPROVED** [review]

**Review by tailingchen — APPROVED** [review]
